docs/development/development-tools.md

```diff
@@ -0,0 +1,119 @@
+# Development Tools
+
+This document provides an overview of two different but related development approaches -
+devcontainers (via Visual Studio Code), and command line tools via a Docker container.
+
+## Visual Studio Code devcontainer
+
+One of your best friends for development may be Visual Studio Code (VSC) and a devcontainer.
+This repo contains a VSC devcontainer all ready for use.
+
+> At the time of writing, devcontainer support for PyCharm is not mature enough to use;
+> too many features are missing. However, in the near future this should also be a very
+> productive development environment.
+
+A devcontainer provides a stable, reproducible development platform. For this repo, it
+is based on the Python 3.11 debian image, the same one used for deployment. In fact, the
+devcontainer should reflect the same OS and Python environment as the deployment and the
+development tools images. This feature is important for reproducibility, and providing
+the least surprise!
+
+There is also support for locally hosted development via VSC, but this developer (EAP)
+does not do this style of Python development.
+
+### Getting Started
+
+1. Ensure you have docker running.
+
+2. Open the project in VSC
+
+    There are a few ways to open a project in VSC:
+
+    - from the repo directory `code .`
+    - from VSC menu:
+    - File > New Window
+    - From the new window:
+        - Click on Explorer in the left-nav, then click on Open Folder
+        - or
+        - Selecte File > Open Folder
+    - Select the folder containing the repo (project)
+    - Click Open
+
+3. Start the devcontainer.
+
+    - press the keys `Shift` - `Command` - `P` (macOS) to open the command palette
+    - start typing "Dev Container: Reopen Folder in Container"
+    - when you see it appear, click on it
+    - the image will build, the container will start
+
+4. Open a terminal
+
+    If a terminal is not already open, open the built-in terminal with `Control` - `~`
+    (that is a tilde)
+
+5. Depending on the docker installation, you may need to grant access. If you don't see
+   git things enabled, click on the Git tool, and follow the instructions.
+
+Now you should treat this just like a local environment. The Python dependencies will
+already be installed. If you have changes to them, however, you may simply make the
+change to the requirements file and then reinstall them from the VSC devcontainer
+terminal.
+
+### Running Tools
+
+All of the tools available for running from the host via docker are also directly
+available within the devcontainer. I still often run them from a host termianl anyway, as
```

Comment:
typo

Reply:
Yeah, fixing typos in that file. I didn't have the spellchecker running so those fingers do slip...